Dockerfile

```diff
@@ -24,23 +24,23 @@ RUN apt-get update &&\
                        python3-wheel \
                        graphviz \
                        libhdf5-dev \
+                       python3.8 \
+                       python3.8-dev \
+                       python3.8-distutils \
                        swig &&\
-    add-apt-repository -y ppa:deadsnakes/ppa &&\
-    apt install -y python3.7 \
-                   python3.7-dev &&\
+    apt-get clean &&\
+    ln -s /usr/bin/python3.8 /usr/local/bin/python &&\
+    ln -s /usr/bin/python3.8 /usr/local/bin/python3 &&\
     curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py &&\
-    python3.7 get-pip.py &&\
+    python3 get-pip.py &&\
     rm get-pip.py &&\
-    ln -s /usr/bin/python3.7 /usr/local/bin/python &&\
-    ln -s /usr/bin/python3.7 /usr/local/bin/python3 &&\
-    apt-get clean &&\
     # best practice to keep the Docker image lean
     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 WORKDIR /src
 
 # Install essential Python packages
-RUN pip3 --no-cache-dir install \
+RUN python3 -m pip --no-cache-dir install \
```

Comment:
does this now work as `pip3`? If not, we should relink `pip3` (and preferably `pip`) to work and target the newly installed Python 3.8.

Reply:
`pip`, `pip3`, `python -m pip` and `python3  -m pip` all use the correct `pip`. From what I read it is generally recommended to use `python -m pip` instead of just `pip` because in the former case you can be sure that you are using the right `pip`.

Reply:
ok as long as `pip` within Docker correctly use the right `pip` then I'm fine.